### PR TITLE
Mirrormorph and Basic Ability Card

### LINKED ISSUE #4080: Downfall Card Implementation Status
Here's the full list:

- [x] Isolation
- [x] Demolisher
- [x] Chisel
- [x] Stargate
- [x] Utae
- [x] Climactic Showdown
- [x] Fencer Fueno
- [x] The Nihilist
- [x] Trickster Taka
- [x] Az McCaffrey: Mechanical Prodigy
- [x] Always Have a Backup Plan
- [x] Blueberry Diesel
- [x] Flip Switch
- [x] Lucky Charm
- [x] Masterwork (v37)
- [x] Bukhgalter
- [x] "Baklan" Bochkin
- [x] The Class Act
- [x] Lat: Ethical Freelancer
- [x] In The Groove
- [x] Khusyuk
- [x] Spec Work
- [x] Supercorridor
- [x] Gauss
- [x] Pelangi
- [x] Rezeki
- [x] The Artist
- [x] Direct Access
- [x] Rejig
- [x] Whistleblower
- [ ] MirrorMorph: Endless Iteration
- [x] Architect Deployment Test
- [x] Calvin B4L3Y
- [x] Nanoetching Matrix
- [x] Hagen
- [x] Fully Operational
- [x] Red Level Clearance
- [x] Cold Site Server
- [x] Hyoubu Institute: Absolute Clarity
- [x] Project Yagi-Uda
- [x] Sting!
- [x] Public Health Portal
- [x] Storgotic Resonator
- [x] Saisentan
- [x] Complete Image
- [x] Letheia Nisei
- [x] Remastered Edition
- [x] Daily Quest
- [x] Tiered Subscription
- [x] Congratulations!
- [x] Loot Box
- [x] Focus Group
- [x] Game Over
- [x] Increased Drop Rates
- [x] Divested Trust
- [x] SDS Drone Deployment
- [x] Roughneck Repair Squad
- [x] Afshar
- [x] Sandstone
- [x] Trebuchet
- [x] Secure and Protect
- [x] Reduced Service
- [x] Vulnerability Audit
- [x] CSR Campaign
- [x] Rime

No linked PRs cuz they all happened off-screen.

### PR DESCRIPTION
**Pull request by lostgeek**
Reworked Basic Actions to be abilities on a single card that is found under `(:basic-action-card side @state)`. There are still some things to be done in a future PR:

- Add the click cost to the basic action for installing, playing and initiating runs
  - This makes it necessary to rework how `cost-str` is passed through the different parts to generate the correct message
- Change `:msg` to take a map with options to remove the card from the message, to avoid messages like "Runner uses Runner Basic Action Card to do X"

Closes #4080

### CODE REVIEW

File: src/clj/game/cards/basic.clj [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+               {:label "Install 1 program, resource, or piece of hardware from the grip"
+                :async true
+                :effect (req (let [target-card (first targets)]
+                               (runner-install state side (make-eid state {:source :action :source-type :runner-install})
+                                               target-card {:base-cost [:click 1]})))}
+               {:label "Play 1 event"
+                :async true
+                :effect (req (let [target-card (first targets)]
+                               (play-instant state side (make-eid state {:source :action :source-type :play})
+                                             target-card {:base-cost [:click 1]})))}
```
Comment (NoahTheDuke) [inline src/clj/game/cards/basic.clj]: You know, I don't think we need this anymore seeing as we now have these abilities. Better to make it just like other abilities and remove the whole `base-cost` hack.

Comment (lostgeek) [reply src/clj/game/cards/basic.clj]: I left this in to display the correct messages. In the long run I'd like to remove all `:base-cost` arguments.

File: src/clj/game/cards/basic.clj [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                             (swap! state update-in [:stats side :click :credit] (fnil inc 0))
+                             (trigger-event state side :corp-click-credit)
+                             (play-sfx state side "click-credit"))}
+               {:label "Draw 1 card"
+                :cost [:click]
+                :msg "draw 1 card"
+                :async true
+                :effect (req (wait-for (trigger-event-simult state side (make-eid state eid) :pre-corp-click-draw nil nil)
+                                       (trigger-event state side :corp-click-draw (->> @state side :deck (take 1)))
+                                       (draw state side)
```
Comment (NoahTheDuke) [inline src/clj/game/cards/basic.clj]: This should be the final line of the ability and should have the `eid` passed in.

File: src/clj/game/cards/basic.clj [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+               {:label "Trash 1 resource if the Runner is tagged"
+                :cost [:click 1 :credit 2]
+                :req (req tagged)
+                :prompt "Choose a resource to trash"
+                :msg (msg "trash " (:title target))
+                :choices {:req (req (if (and (seq (filter (fn [c] (untrashable-while-resources? c)) (all-active-installed state :runner)))
+                                             (> (count (filter resource? (all-active-installed state :runner))) 1))
+                                      (and (resource? target) (not (untrashable-while-resources? target)))
+                                      (resource? target)))}
+                :effect (effect (trash target))}
```
Comment (NoahTheDuke) [inline src/clj/game/cards/basic.clj]: Should pass the `eid` in, and this ability should be `:async true`

File: src/clj/game/cards/basic.clj [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+                :effect (req (gain-credits state side 1 :runner-click-credit)
+                             (swap! state update-in [:stats side :click :credit] (fnil inc 0))
+                             (trigger-event state side :runner-click-credit)
+                             (play-sfx state side "click-credit"))}
+               {:label "Draw 1 card"
+                :cost [:click]
+                :msg "draw 1 card"
+                :effect (req (wait-for (trigger-event-simult state side (make-eid state eid) :pre-runner-click-draw nil nil)
+                                       (trigger-event state side :runner-click-draw (->> @state side :deck (take 1)))
+                                       (draw state side)
```
Comment (NoahTheDuke) [inline src/clj/game/cards/basic.clj]: Same as above

File: src/clj/game/cards/basic.clj [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                                       (trigger-event state side :corp-click-draw (->> @state side :deck (take 1)))
+                                       (draw state side)
+                                       (swap! state update-in [:stats side :click :draw] (fnil inc 0))
+                                       (play-sfx state side "click-card")))}
+               {:label "Install 1 agenda, asset, upgrade, or piece of ice from HQ"
+                :async true
+                :req (req (not-empty (:hand corp)))
+                :effect (req (let [target-card (first targets)
+                                   server (second targets)]
+                               (corp-install state side (make-eid state {:source server :source-type :corp-install})
```
Comment (NoahTheDuke) [inline src/clj/game/cards/basic.clj]: Should pass the `eid` in instead of making a new one.

File: src/clj/game/cards/basic.clj [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+                                             target-card server {:base-cost [:click 1] :action :corp-click-install})))}
+               {:label "Play 1 operation"
+                :async true
+                :req (req (not-empty (:hand corp)))
+                :effect (req (let [target-card (first targets)]
+                               (play-instant state side (make-eid state {:source :action :source-type :play})
```
Comment (NoahTheDuke) [inline src/clj/game/cards/basic.clj]: Same eid comment as above.

File: src/clj/game/cards/basic.clj [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+                :effect (req (wait-for (trigger-event-simult state side (make-eid state eid) :pre-runner-click-draw nil nil)
+                                       (trigger-event state side :runner-click-draw (->> @state side :deck (take 1)))
+                                       (draw state side)
+                                       (swap! state update-in [:stats side :click :draw] (fnil inc 0))
+                                       (play-sfx state side "click-card")))}
+               {:label "Install 1 program, resource, or piece of hardware from the grip"
+                :async true
+                :req (req (not-empty (:hand runner)))
+                :effect (req (let [target-card (first targets)]
+                               (runner-install state side (make-eid state {:source :action :source-type :runner-install})
```
Comment (NoahTheDuke) [inline src/clj/game/cards/basic.clj]: Same eid comment as above.

File: src/clj/game/cards/basic.clj [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+                                               target-card {:base-cost [:click 1]})))}
+               {:label "Play 1 event"
+                :async true
+                :req (req (not-empty (:hand runner)))
+                :effect (req (let [target-card (first targets)]
+                               (play-instant state side (make-eid state {:source :action :source-type :play})
```
Comment (NoahTheDuke) [inline src/clj/game/cards/basic.clj]: Same eid comment as above.

File: src/clj/game/cards/basic.clj [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+                                             target-card {:base-cost [:click 1]})))}
+               {:label "Run any server"
+                :async true
+                :effect (effect (make-run (make-eid state) target nil nil {:click-run true}))}
```
Comment (NoahTheDuke) [inline src/clj/game/cards/basic.clj]: Same eid comment as above.

**Pull request merged**